server/topic.go

```diff
@@ -949,20 +956,26 @@ func (t *Topic) handleBroadcast(msg *ServerComMessage) {
 				}
 
 			} else {
-				// Check if the user has Read permission
-				if !t.userIsReader(pssd.uid) {
+				// Check if the user has Read permission or is channel reader.
+				if !t.userIsReader(pssd.uid) && !pssd.isChanSub {
 					continue
 				}
 
 				// Don't send key presses from one user's session to the other sessions of the same user.
-				if msg.Info != nil && msg.Info.What == "kp" && msg.Info.From == pssd.uid.UserId() {
+				if !pssd.isChanSub && msg.Info != nil && msg.Info.What == "kp" && msg.Info.From == pssd.uid.UserId() {
 					continue
 				}
 			}
 		}
 
 		// Topic name may be different depending on the user to which the `sess` belongs.
 		t.maybeFixTopicName(msg, pssd.uid)
+
+		// Send channel messages anonymously.
+		if pssd.isChanSub && msg.Data != nil {
+			msg.Data.From = ""
```

Comment:
In this case, we don't want multiple posters on the channel?
Telegram, AFAIK, displays posters' names in channels.

Reply:
I checked: telegram-web always shows channel name as a poster name, telegram-android does not show the poster name at all. I did not check iOS. 
I could set `From` to the same value as the channel name but that seems redundant. Clients can do it if necessary.
I dislike the idea of showing actual poster name in channels because the name is not important. It's kind of like a newspaper.

Reply:
https://telegram.org/tour/channels

> Unlike Telegram Groups, channels show the name and photo of the channel next to messages – instead of the person who posted them.

Reply:
Ah, I see. I think it's the case on Telegram web client only.
E.g. https://t.me/s/my_ali_bot (see the bottom right corner of messages).